pkg/kubectl/cmd/set/helper.go

```diff
@@ -139,7 +139,14 @@ func CalculatePatch(patch *Patch, encoder runtime.Encoder, mutateFn patchFn) boo
 		return false
 	}
 
-	// TODO: should be via New
+	// Create the versioned struct from the type defined in the restmapping
+	// (which is the API version we'll be submitting the patch to)
+	_, err := legacyscheme.Scheme.New(patch.Info.Mapping.GroupVersionKind)
+	if err != nil {
+		patch.Err = err
+		return true
+	}
+
 	versioned, err := patch.Info.Mapping.ConvertToVersion(patch.Info.Object, patch.Info.Mapping.GroupVersionKind.GroupVersion())
```

Comment:
I'd actually be interested to see if `patch.Info.VersionedObject` could be used instead, removing the need to call ConvertToVersion here or to create versionedEncoder above